lib/zendesk_api/resources.rb

```diff
@@ -857,4 +857,16 @@ class Variant < ZendeskAPI::Resource
       has_many Variant
     end
   end
+
+  class PushNotificationDevice < Resource
```

Comment:
`DataResource` is the base class, `Resource` inherits from it

`Resource` is a `DataResource` with full CRUD (POST for create, GET for read, PUT for update, DELETE for destroy)

There's also `DeleteResource` but that supports single `DELETE`, so you probably don't want that as the superclass either.

https://github.com/zendesk/zendesk_api_client_rb/blob/master/lib/zendesk_api/resource.rb#L142-175